src/bin2pat/bin2pat.cpp

```diff
@@ -81,6 +83,37 @@ void processArgs(
 				return;
 			}
 		}
+		else if (args[i] == "-l" || args[i] == "--list") {
+			// Ensure -l --list is not the last thing in args
+			if (&args[i] == &args.back()) {
```

Comment:
Here, I think the code would be more readable if we just compared `i` with the size of `args`:
```cpp
if (i == args.size() - 1) {
```
The comparison of addresses feels very unnatural and kept me thinking really hard whether the check is actually correct (I am still not entirely convinced that it is correct). What do you think?

Reply:
The reason I used a comparison of addresses here is because it is absolute. I don't see how this could potentially fail and while I didn't go to the ends of the Earth to break it, it did work.

Using a comparison of addresses ensures thst, at the time the program counter reaches that section of code, that the argument being read is currently the last argument in args.

I can change it for readability reasons and the difference breaks down to only one instruction due to the subtraction.

The only reason I did not use your above suggestion is because it is comparing the current index count to the size of args in relation to it's index as opposed to comparing if the current argument being evaluated is the final argument in args.